src/main/java/org/cqfn/astranaut/core/algorithms/Subtree.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2024 Ivan Kniazkov
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cqfn.astranaut.core.algorithms;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.cqfn.astranaut.core.Node;
+
+/**
+ * Subtree.
+ *
+ * @since 1.1.4
+ */
+public class Subtree {
+    /**
+     * The root of the subtree.
+     */
+    private final SubtreeNode root;
+
+    /**
+     * The set of nodes for this subtree.
+     */
+    private final Set<Node> nodes;
+
+    /**
+     * The mapping of the nodes' children for this subtree.
+     */
+    private final Map<Node, List<Integer>> mapping;
+
+    /**
+     * Constructor.
+     *
+     * @param root The root of the subtree
+     * @param nodes The set of nodes in this subtree
+     */
+    @SuppressWarnings("PMD.ConstructorOnlyInitializesOrCallOtherConstructors")
+    public Subtree(final Node root, final Set<Node> nodes) {
```

Comment:
I think here should be only one parameter, namely, source tree.
Don't perform calculations in constructors.
Expected interface should be like this:

```
Subtree subtree = new Subtree(source);
Node first = subtree.create(Arrays.asList(...)); // first subtree created from the source node and set of nodes
Node second = subtree.create(Arrays.asList(...));
```
Thus, using one subtree object, we should be able to create one or more subtrees

Reply:
Changed